staging/src/k8s.io/client-go/rest/warnings.go

```diff
@@ -43,33 +51,70 @@ var (
 //   - NoWarnings suppresses warnings.
 //   - WarningLogger logs warnings.
 //   - NewWarningWriter() outputs warnings to the provided writer.
+//
+// logcheck:context // SetDefaultWarningHandlerWithContext should be used instead of SetDefaultWarningHandler in code which supports contextual logging.
 func SetDefaultWarningHandler(l WarningHandler) {
+	defaultWarningHandlerLock.Lock()
+	defer defaultWarningHandlerLock.Unlock()
+	defaultWarningHandler = warningLoggerNopContext{l: l}
```

Comment:
Nit:
```suggestion
	SetDefaulthandlerWithContext(warningLoggerNopContext{l: l})
```

Reply:
There are some more below that look similar.

Reply:
I'll fix this tomorrow, then @aojea and/or @sbueringer can re-add the LGTM.

Reply:
Fixed. The one below in `getDefaultWarningHandler` (if that's the one) is different.